packages/server/src/keyvalue/store.ts

```diff
@@ -0,0 +1,83 @@
+import { OperationOutcomeError, badRequest, notFound } from '@medplum/core';
+import { AuthenticatedRequestContext, getAuthenticatedContext } from '../context';
+import { getRedis } from '../redis';
+
+export const MAX_KEY_LENGTH = 255;
+export const MAX_VALUE_LENGTH = 8192;
+export const MAX_ITEMS = 10;
+
+/**
+ * Gets the value for the given key from the keyvalue store.
+ * @param key - The key to get the value for.
+ * @returns The value for the given key.
+ */
+export async function getValue(key: string): Promise<string> {
+  validateKey(key);
+  const ctx = getAuthenticatedContext();
+  const value = await getRedis().hget(getCacheKey(ctx), key);
+  if (value === null) {
+    throw new OperationOutcomeError(notFound);
+  }
+  return value;
+}
+
+/**
+ * Sets the value for the given key in the keyvalue store.
+ * @param key - The key to set the value for.
+ * @param value - The value to set.
+ */
+export async function setValue(key: string, value: string): Promise<void> {
+  validateKey(key);
+  if (typeof value !== 'string') {
+    throw new OperationOutcomeError(badRequest('Invalid value type'));
+  }
+  if (value.length > MAX_VALUE_LENGTH) {
+    throw new OperationOutcomeError(badRequest('Value too long'));
+  }
+  const ctx = getAuthenticatedContext();
+  const cacheKey = getCacheKey(ctx);
+  const redis = getRedis();
+  const exists = await redis.hexists(cacheKey, key);
+  if (!exists) {
+    const length = await redis.hlen(cacheKey);
+    if (length >= MAX_ITEMS) {
+      throw new OperationOutcomeError(badRequest('Max items exceeded'));
+    }
+  }
+  await redis.hset(cacheKey, key, value);
```

Comment:
There's a small race condition here where someone could spam a lot of set requests at the same time and have more than `MAX_ITEMS` wind up being set. We could avoid it by obtaining/releasing a lock around the operation, but I'm not sure if it's worth the complexity.

Reply:
Acknowledged.  If someone gets 11 (or even 20) keys, that's not going to kill us.